### PR TITLE
Components: replace `TabPanel` with `Tabs` in the editor Document Overview sidebar

### PR DESCRIPTION
**Pull request by chad1008**
## What?
Replaces the legacy TabPanel component with the new Tabs component.

## Why?
Part of the work outlined in https://github.com/WordPress/gutenberg/issues/52997

## How?
`TabPanel` is replaced by `Tabs` and its sub-components.

The biggest change in terms of actual behavior is the order of the tab stops. Previously, the close button had to be rendered outside of (specifically before) `TabPanel` and then placed next to the tabs via CSS. With `Tabs` we can actually compose the sidebar with the button right where we want it. This changes the flow of focus when tabbing into the sidebar:

<table>
<tr>
<th>Before</th>
<th>After</th>
<tr>
 <td>
  <ol>
   <li>Close Button</li>
   <li>Active Tab</li>
   <li>The currently displayed `tabpanel`</li>
   <li> First focusable element within the current `tabpanel`</li>
  </ol>
 </td>
 <td>
  <ol>
   <li>Active Tab</li>
   <li>Close Button</li>
   <li>The currently displayed `tabpanel`</li>
   <li> First focusable element within the current `tabpanel`</li>
  </ol>
 </td>
</table>

This matches what the Editor Settings sidebar on the other side of the screen does, but I wanted to confirm this is an acceptable change from an a11y point of view, for both sidebars. cc @andrewhayward @alexstine 

## Notes
1. There are some visual changes at the moment regarding hover effects and text alignment on the tabs. These are being discuss/addressed in #57121
2. There is currently one failing e2e test, which is a direct result of the change in tabstops mentioned above. Before this merges, I'll make sure we update either the test or the tab flow as necessary.

## Testing Instructions
1. Create a new post and open the Document Overview sidebar
2. Test the **Document Overview** and **Outline** tabs, confirming they work as expected and display the correct data
3. Open and close the sidebar to confirm there are no errors.
4. Test the shortcut keys (`alt+shift+o` on Windows, and `control+option+o` on Mac) to confirm it opens and closes the sidebar properly

This sidebar has some specific focus behavior built-in. The following tests are to double check we aren't breaking any of that.
tl;dr:
- When closed, the shortcut key should open the sidebar.
- When open, if the sidebar doesn't have focus the shortcut should apply focus to the correct element in the sidebar.
- When open, if the sidebar does have focus, the shortcut should close the sidebar
- When the shortcut closes the sidebar, focus should land on the Document Overview sidebar toggle button

**On a post with no blocks:**
1. Click to place focus on the canvas
2. Press the Document Overview shortcut keys
3. Document Overview should open, focus should remain on the editor canvas
4. Press the shortcut again, focus should move to the first tabbable item in the sidebar (ie, the List View tab)
5. Press the shortcut one more time, and the sidebar should close
6. Confirm that when the sidebar closes, the focus goes to the sidebar toggle button
7.  Open the Document Overview again (either by clicking or by shortcut, it doesn't matter which)
8. Click any element in the sidebar to focus it
9. Press the shortcut keys, and the sidebar should close
10. Open the sidebar, and then click the canvas to move focus outside the sidebar
11. Press the shortcut keys, and confirm focus moves to the first tabbale item (which should still be the List View tab)
12. Press the shortcut keys again, the sidebar should close

**On a post with blocks:**
1. Click to place focus on the canvas
2. Press the Document Overview shortcut keys
3. Document Overview should open, focus should go straight to the first block listed in the sidebar
4. Pressing the shortcut keys again should close the sidebar
5.  Confirm that when the sidebar closes, the focus goes to the sidebar toggle button
6. Open the sidebar, and then click the canvas to move focus outside the sidebar
7. Press the shortcut keys, and confirm focus moves to the first block in the list
8. Press the shortcut keys again, the sidebar should close

### Testing Instructions for Keyboard
1. Create a new post
2. Press `` ctrl+` `` four times to navigate to the top toolbar region
3. Press [Tab] to focus the first toolbar button
4. [ArrowRight] to the **Document Overview** button
5. Press [Enter/Return/Space] to action the button and open the sidebar
6. Press [Tab] until you've traversed the top toolbar
7. When focus lands on the **List View** tab, confirm that arrow keys can be used to navigate between tabs
8. Select a tab with [Enter/Return/Space]
9. Confirm that [Tab] moves focus to the **Close** button
10. Confirm that [Tab] again moves focus to the canvas, because there's nothing in the list view/outline
11. Close the sidebar by pressing the shortcut key twice
12. Add some text to the default paragraph block
13. Use `` ctrl+` `` to once again navigate to the top toolbar
14. [Tab] will now focus the **Document Overview** toggle
15. Open the sidebar, confirm that focus jumps to the first block in the list

### CODE REVIEW

**Comment by andrewhayward** [comment]
Thanks for working on this @chad1008, and for the thorough testing instructions!

> Document Overview should open, focus should remain on the editor canvas

Some of the interaction flow feels a bit weird to me (for example, not moving focus), but I'll leave that alone, as it's not introduced here.

### Focus order

> The biggest change in terms of actual behavior is the order of the tab stops [...] This matches what the Editor Settings sidebar on the other side of the screen does, but I wanted to confirm this is an acceptable change from an a11y point of view, for both sidebars.

For me, the 'before' behaviour, where the close button is before the tab control in the focus order, is more _correct_ than the 'after' behaviour, and we'd ideally change the Editor Settings sidebar behaviour to match.

I'm not aware of any specific guidance regarding this particular scenario, where other controls are at play, but in general, I would expect the tab panel to immediately follow the tabs in the focus order, if at all possible...

> When keyboard focus is on a `tablist`, or a tab within the `tablist`, the <kbd>Tab</kbd> key should be programmed to move from the focused tab — which may or may not be the selected tab — to the `tabpanel` which represents the currently selected tab.
> — _([MDN: Tabpanel role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tabpanel_role))_

> When focus moves into the tab list, <kbd>Tab</kbd> places focus on the active tab element. When the tab list contains the focus, <kbd>Tab</kbd> moves focus to the next element in the page tab sequence outside the tablist, which is the tabpanel unless the first element containing meaningful content inside the tabpanel is focusable.
> — _([W3: Tabs pattern keyboard interaction](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/#keyboard_interaction))_

### Tab selection

I also wanted to briefly mention tab selection.

> Select a tab with [Enter/Return/Space]

Is there a reason why selection doesn't follow focus here?

> It is recommended that tabs activate automatically when they receive focus as long as their associated tab panels are displayed without noticeable latency.
> — _([W3: Tabs pattern keyboard interaction](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/#keyboard_interaction))_

I don't want to assume we haven't had the conversation, but [it doesn't feel like there's a good reason not to](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_selection_follows_focus). Not a blocker by any means.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 1]
```diff
@@ -109,62 +115,51 @@ export default function ListViewSidebar( { listViewToggleElement } ) {
...
-					{
-						name: 'outline',
-						title: _x( 'Outline', 'Post overview' ),
-						className: 'edit-post-sidebar__panel-tab',
-					},
-				] }
 			>
-				{ ( currentTab ) => (
+				<Tabs
+					selectOnMove={ false }
```
Comment (ciampo) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Echoing @andrewhayward 's question — is there a reason for this setting?

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 2]
```diff
@@ -109,62 +115,51 @@ export default function ListViewSidebar( { listViewToggleElement } ) {
...
+						className="edit-post-editor__document-overview-panel__close-button"
+						icon={ closeSmall }
+						label={ __( 'Close' ) }
+						onClick={ closeListView }
+					/>
+
 					<div
 						className="edit-post-editor__list-view-container"
 						ref={ listViewContainerRef }
 					>
```
Comment (ciampo) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: I wonder if we could get rid of this `div` and somehow pass the relevant classname and ref to the tab panels (or find an alternative solution)?

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 3]
```diff
@@ -109,62 +115,51 @@ export default function ListViewSidebar( { listViewToggleElement } ) {
...
 			className="edit-post-editor__document-overview-panel"
 			onKeyDown={ closeOnEscape }
 			ref={ sidebarRef }
 		>
-			<Button
-				className="edit-post-editor__document-overview-panel__close-button"
-				icon={ closeSmall }
-				label={ __( 'Close' ) }
-				onClick={ closeListView }
-			/>
```
Comment (ciampo) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: As @andrewhayward pointed out, `Button` used to come before the tablist in DOM order and therefore in tab order too. 

Can we move it back to before the tablist? If necessary, we should be able to play around with CSS and DOM structure to write a cleaner layout (ie. no position absolute for the close button), thanks to the flexibility that `Tabs` offers with its subcomponents

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 1 above)
Comment (chad1008) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: My reason for including it was to maintain parity with existing behavior, but after thinking about this question I wish I'd looked a little more into why it was previously set up this way under `TabPanel`.

Disabling automatic selection was [first introduced](https://github.com/WordPress/gutenberg/pull/50199#discussion_r1181006346) when `TabPanel` was implemented. It was used to [avoid the sidebar's focus behavior from switching tabs in ways that would be disruptive or unexpected for users](https://github.com/WordPress/gutenberg/pull/50199#issuecomment-1568472931).

All of that said, no matter how I test it I can't actually replicate a case where I'm looking at one tab, and the focus behavior drags me to the other. If I select the **Outline** tab, then focus the canvas, and then hit the shortcut keys, focus goes to my current tab, which is **Outline**. If focus went to **List View** and it were automatically selected, that could be a little disorienting, but it doesn't seem to be possible.

Theory: The implementation of `TabPanel` in this sidebar predates that component getting updated with `Ariakit` internals. The reason focus lands on the currently selected tab is because of how `Ariakit` applies a negative `tabindex` to all tabs that aren't currently selected.

I suspect pre-`Ariakit` `TabPanel` didn't have a similar implementation, meaning the first tab, even unselected, could have received focus. Without the prop in question, that would cause a potentially jarring change. Rolling back to the old `TabPanel` without `Ariakit` is kind of a pain on my setup, so I'm going to roll with this assumption being accurate.

So to sum up, I think this was a necessary prop at the time, but since the arrival of `Ariakit` we probably no longer need it. Thanks for asking the question I should have asked myself at the outset 😉

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 3 above)
Comment (chad1008) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: @andrewhayward  & @ciampo Thank you both for the input regarding the order of the elements and tab flow! I've moved it back before the `tablist`, and updated the related styles in a454774478 so we're no longer relying on `position: absolute`. I did end up adding a wrapper `div` which I'd have preferred to avoid, but having a `flex` container made life a lot easier.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 4]
```diff
@@ -109,62 +115,52 @@ export default function ListViewSidebar( { listViewToggleElement } ) {
...
-			<TabPanel
+			<div
```
Comment (ciampo) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Is keeping this `div` around necessary?

Comment (chad1008) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Hm... perhaps not! It looks like it's really only here to carry the ref! The style associated with this classname forces 100% height, but I think that's just to keep the div from collapsing and shrinking the drop zone.

The ref is used to handle the focus behavior when there aren't any blocks in the list view to focus. If we put that ref onto the `tablist` (and rename it!), that behavior should still work normally. We then don't need the `div`, the classname, or the related style. Nice, thank you!

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 2 above)
Comment (chad1008) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: I think we can, if we just pass the classname to the `tabpanel`s themselves they styles shouldn't be negatively impacted. The ref looks, unless I'm mistaken, like it's only relevant to the List View, so we can pass the ref there directly.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js [hunk 5]
```diff
@@ -111,62 +116,49 @@ export default function ListViewSidebar() {
...
+							{ _x( 'List View', 'Post overview' ) }
+						</Tabs.Tab>
+						<Tabs.Tab tabId="outline">
+							{ _x( 'Outline', 'Post overview' ) }
+						</Tabs.Tab>
+					</Tabs.TabList>
+				</div>
+
+				<Tabs.TabPanel
+					ref={ listViewContainerRef }
```
Comment (ciampo) [inline packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Just noting that previously `listViewContainerRef` was passed to all tabpanels, while now it's only passed to the list view tabpanel.

`listViewContainerRef` is used for 3 purposes:
- `focusOnMountRef`: used to move focus when the component mountsl. Changes from this PR shouldn't change a regression, since the default tab is hardcoded to be `'list-view'`, and therefore the outline tabpanel doesn't need a ref
- `listViewRef`: used to handle focus inside the `'list-view'` tab. Again, this shouldn't be needed by the outline tab
- `setDropZoneElement`: used to set the HTML drop zone as a callback ref. This also doesn't seem related to the outline tab.

In short, it seems OK that we're passing `listViewContainerRef` only to the list view tab panel.

**Review by ciampo — COMMENTED** [review]
Code changes look good. I always followed the testing instructions and couldn't spot any regressions 🚀 

I'd be more comfortable if @andrewserong also took a look before approving, given his extensive work on the list view in the past months.

File: packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js (hunk 5 above)
Comment (chad1008) [reply packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js]: Thank you for adding more context around this change for anyone looking at it later.

**Review by andrewserong — COMMENTED** [review]
> I'd be more comfortable if @andrewserong also took a look before approving

Thanks for the ping, and for the work here!

> There are some visual changes at the moment regarding hover effects and text alignment on the tabs. These are being discuss/addressed in https://github.com/WordPress/gutenberg/pull/57121

I see this is now centering the labels for the buttons. I think these should be left aligned as on `trunk` unless this is an intentional design decision? To my eyes it looks a bit unbalanced having them be centered as other panels (like the block or template inspector controls) are left aligned.

| Trunk | This PR |
| --- | --- |
| image | image |

While it's fairly subtle, because the list view is a prominent part of the UI, I think it'd be good to settle on the desired design here before landing the PR.

In terms of usability, the keyboard behaviour is mostly working for me. The main issue I ran into is if you press the Right arrow key to go to Outline and then the Left arrow key to go back to the list view, then the focus behaviour appears to move the focus to the first selected list view item. Is that intentional? It feels like a bug to me, as I believe that focusing behaviour is only intended to run when the list view is initially opened. If it is intentional, it might be worth checking with @alexstine here, too, as I could imagine it being confusing when moving between panels.

https://github.com/WordPress/gutenberg/assets/14988353/58c4101c-dea8-45e6-855e-0bbe1fee116c

I'll be AFK after tomorrow, so apologies if I don't get back to this PR quickly! If it's still open in the new year, I'm happy to give it a re-review then 🙂

**Comment by ciampo** [comment]
> I see this is now centering the labels for the buttons.

Good point! https://github.com/WordPress/gutenberg/pull/57275 has just been merged, and it contains the style tweaks to change the tabs text alignment :) 

> The main issue I ran into is if you press the Right arrow key to go to Outline and then the Left arrow key to go back to the list view, then the focus behaviour appears to move the focus to the first selected list view item. Is that intentional? It feels like a bug to me,

That also feels like a bug, we should look more into it.

> I'll be AFK after tomorrow, so apologies if I don't get back to this PR quickly! 

@chad1008 is also going to be AFK.

Let's resume work on this after the holidays

**Comment by andrewserong** [comment]
> Let's resume work on this after the holidays

Sounds good. Happy to help with re-reviews then!

**Comment by chad1008** [comment]
And we're back! Thanks for jumping in on this one @andrewserong, very good catch.

The reason this is happening is because I enabled `focusOnMove`. The previous `TabPanel` implementation intentionally had this disabled to account for this sidebar's specific focus behavior... when putting the PR together I thought the new focus handling introduced by `Tabs` made this unnecessary, but you've found an aspect to it that I missed.

Turning `focusOnMove` off again restores keyboard nav behavior currently present on `trunk`

1. open sidebar
2. select block
3. click on the **List View** tab
4. arrow key back and forth to your heart's content
5. when you actually select the **List View** tab via [Return/Enter/Space] focus will then transition to the currently selected block, as it's supposed to whenever `List View` activates.

Sorry for overlooking that, and thank you again for spotting it.

**Review by andrewserong — COMMENTED** [review]
Thanks for the updates @chad1008! That's resolved the keyboard behaviour for me 👍

In re-testing, unfortunately I think this PR appears to break the windowing logic for the list view, which controls the showing and hiding of list view items on very long posts. On trunk, if I scroll up and down the list view, items and shown and hidden. With this PR applied, the updates aren't occurring:

## Trunk

https://github.com/WordPress/gutenberg/assets/14988353/8808ae9a-77c6-4167-b300-7a30f66f3610

## This PR

https://github.com/WordPress/gutenberg/assets/14988353/5eea0974-e8b9-46aa-8d35-0026163c265c

For quickly testing very long posts, I'll either use a lorem ipsum generator or copy & paste [an old book from the "other" Gutenberg](https://www.gutenberg.org/cache/epub/345/pg345-images.html) 😄.

I'm not too sure what the cause is, but in case it helps digging into it, the list view calls [`useFixedWindowList` here](https://github.com/WordPress/gutenberg/blob/f927f2d348be9e23739496ff69c49b1fae2cfa7b/packages/block-editor/src/components/list-view/index.js#L244), and `useFixedWindowList` [is defined here](https://github.com/WordPress/gutenberg/blob/f27369510c1a4702860a32b959f4e62d5bf89972/packages/compose/src/hooks/use-fixed-window-list/index.js#L41).

I wonder if something about the `Tabs.TabPanel` could be interfering with the `useFixedWindowList`'s checking for a scroll container? I notice with this PR applied that [the line that calls `getScrollContainer`](https://github.com/WordPress/gutenberg/blob/f27369510c1a4702860a32b959f4e62d5bf89972/packages/compose/src/hooks/use-fixed-window-list/index.js#L63) returns the overall html document instead of the list view container, which is unexpected:

image

Another potential thing to look into if it isn't `Tabs.TabPanel` — could any of the CSS changes in this PR affect the `getScrollContainer` behaviour?

One other thing to keep in mind with changes to this component is that @youknowriad  has opened a PR (https://github.com/WordPress/gutenberg/pull/57467) to propose moving the component to live within the `editor` package so that it can be shared between both the post and site editors. So, whichever PR lands first, the other one will likely need some rebasing 🙂

Let me know how you go, happy to help with continued testing!

**Comment by ciampo** [comment]
> has opened a PR (#57467) to propose moving the component to live within the `editor` package so that it can be shared between both the post and site editors. So, whichever PR lands first, the other one will likely need some rebasing 🙂

#57467 got merged, so this PR needs to be rebased. Let's also make sure we keep in mind the differences between post and site editor as highlighted in https://github.com/WordPress/gutenberg/pull/57467#issuecomment-1875110042

**Comment by ciampo** [comment]
> I wonder if something about the Tabs.TabPanel could be interfering with the useFixedWindowList's checking for a scroll container?

@andrewserong , Chad and I debugged the issue and found out that the problem may be caused by `Tabs` not loading with a default selected tab id. This causes the `Tabs` component to have to figure out which tab to show when mounted, which in return causes a delay in rendering the list view. As a consequence, the `scrollHeight` and `clientHeight` properties in [this check](https://github.com/WordPress/gutenberg/blob/e798d9bbd5464a737cfa01b25894a65b8413868f/packages/dom/src/dom/get-scroll-container.js#L22) are initially computed with a value of `0`, which makes the check fail, and causes the `getScrollContainer` function to continue traversing the DOM until it finds the HTML element. Subsequent re-renders would instead successfully pass the check, and therefore return the correct scroll container.

Luckily, there seems to be an easy fix — adding the `initialTabId` prop to `Tabs` should cause the component to render the selected tab immediately, ultimately allowing the `getScrollContainer` function to parse the height correctly.

I will let @chad1008 work on implementing the fix (together with rebasing after https://github.com/WordPress/gutenberg/pull/57467 got merged), and then hopefully we'll be able to merge this PR 🤞

**Comment by andrewserong** [comment]
> Luckily, there seems to be an easy fix — adding the initialTabId prop to Tabs should cause the component to render the selected tab immediately, ultimately allowing the getScrollContainer function to parse the height correctly.

Nice debugging, folks! Let me know when this is ready for another review and I'll give it another spin 🙂

**Comment by chad1008** [comment]
This PR slipped off my radar somehow, but I've made the change mentioned above that should prevent the poorly rendering scroll container for the list view. I've also rebased to incorporate the newest changes from trunk.

Would love a fresh opinion whenever you have a moment @andrewserong!

File: packages/editor/src/components/list-view-sidebar/style.scss [hunk 6]
```diff
@@ -82,3 +80,9 @@
 		color: $gray-700;
 	}
 }
+
+.edit-post-editor__list-view-container {
+	display: flex;
+	flex-direction: column;
+	height: 100%;
+}
```
Comment (andrewserong) [inline packages/editor/src/components/list-view-sidebar/style.scss]: Looks like these lines might be left over from the rebase? There is no longer a `edit-post-editor__list-view-container` classname, but there is a `editor-list-view-sidebar__list-view-container` with the same rules further up in this file 🙂

File: packages/editor/src/components/list-view-sidebar/index.js [hunk 7]
```diff
@@ -108,62 +113,53 @@ export default function ListViewSidebar() {
...
+					tabId="list-view"
+					focusable={ false }
+				>
+					<div className="editor-list-view-sidebar__list-view-panel-content">
+						<ListView dropZoneElement={ dropZoneElement } />
 					</div>
-				) }
-			</TabPanel>
+				</Tabs.TabPanel>
+				<Tabs.TabPanel tabId="outline" focusable={ false }>
```
Comment (andrewserong) [inline packages/editor/src/components/list-view-sidebar/index.js]: Is this missing a container classname? Just wondering if that's the cause of the padding rules for the outline no longer being applied: https://github.com/WordPress/gutenberg/blob/31ad09c10d916a1503d706055577f4267a0704cc/packages/editor/src/components/list-view-sidebar/style.scss#L40

**Review by andrewserong — COMMENTED** [review]
Thanks for the ping, this is looking _so_ close now @chad1008! The list view keyboard behaviour all appears to be working nicely for me now, and the issues we'd uncovered so far all appear to have been resolved (can switch between List View and Outline easily via keyboard, windowing logic is working).

While comparing between trunk and this PR, I noticed that it looks like some of the re-ordering means that the document outline has lost its padding. I've left a comment where I think there might be a classname missing for the document outline? This is the set of rules that don't appear to be applying with this PR active:

https://github.com/WordPress/gutenberg/blob/31ad09c10d916a1503d706055577f4267a0704cc/packages/editor/src/components/list-view-sidebar/style.scss#L40

The result is that the document outline appears flush against the side of the viewport, instead of having a little breathing room (most notable when an `h1` tag is in use):

### Trunk

image

### This PR

image

Perhaps a little clearer when navigating through the outline via keyboard, where the left edge of the outline gets cut off:

| Trunk | This PR |
| --- | --- |
| image | image |

Thanks again for all the detailed work on this one, I know it's a nuanced one!

File: packages/editor/src/components/list-view-sidebar/style.scss (hunk 6 above)
Comment (chad1008) [reply packages/editor/src/components/list-view-sidebar/style.scss]: Shoot, i thought i'd removed that. I noticed it after the rebase didn't go as planned. Thank you!

File: packages/editor/src/components/list-view-sidebar/index.js (hunk 7 above)
Comment (chad1008) [reply packages/editor/src/components/list-view-sidebar/index.js]: ah, yes. and i see where i was misreading things when working on the rebase... well spotted, thanks!

**Comment by chad1008** [comment]
Thanks you as always @andrewserong, both points on your feedback were good catches!

There is still a small visual shift in the placement of the second tab, which comes from the spacing for the close button no longer being an implicitly declared margin. I don't think it looks problematic but happy to tweak it if need be.

I did also add one small change replacing role based selectors with actual class names, based on [some feedback on a different implementation of this component](https://github.com/WordPress/gutenberg/pull/57293#discussion_r1446521142). It shouldn't have any impact on the look of the sidebar.

**Review by andrewserong — COMMENTED** [review]
Thanks for the updates @chad1008, the document outline is looking good now!

> There is still a small visual shift in the placement of the second tab, which comes from the spacing for the close button no longer being an implicitly declared margin. I don't think it looks problematic but happy to tweak it if need be.

That one looks fine to me, I doubt anyone will really notice it 🙂

In re-testing I ran into one blocking issue which is that the horizontal scrollbar appears to no longer be available in deeply nested lists. This is more common in pretty complex templates, but here's an example with a very heavily nested series of Group blocks:

## Trunk

On hover, a horizontal scrollbar is available:

https://github.com/WordPress/gutenberg/assets/14988353/87d22e0e-31ab-44d7-b280-f29a2041c38f

## This PR

No horizontal scrollbar is available:

https://github.com/WordPress/gutenberg/assets/14988353/3fe2dc8c-7b36-4944-a8d8-86a8568b5a39

Other than that, everything else appears to be testing nicely so far! It might be good to give this another rebase, too, so that we can test it in combination with some of the list view drag and drop changes that landed last week. Rebasing locally, it seems to play nicely, though 👍

**Comment by chad1008** [comment]
thank you yet again @andrewserong for testing so thoroughly and finding the issues i'd missed!

Some of the DOM structure was off, but I've addressed that and the scrollbars should be good now!

I've also performed a fresh rebase and everything is testing well for me locally, but let me know that you think!

File: packages/editor/src/components/list-view-sidebar/index.js [hunk 8]
```diff
@@ -108,62 +113,70 @@ export default function ListViewSidebar() {
...
-						title: _x( 'List View', 'Post overview' ),
-						className: 'editor-list-view-sidebar__panel-tab',
-					},
-					{
-						name: 'outline',
-						title: _x( 'Outline', 'Post overview' ),
-						className: 'editor-list-view-sidebar__panel-tab',
-					},
-				] }
+				// The initial tab value is set explicitly to avoid a an initial
```
Comment (andrewserong) [inline packages/editor/src/components/list-view-sidebar/index.js]: Tiny typo 😄

```suggestion
				// The initial tab value is set explicitly to avoid an initial
```

**Review by andrewserong — APPROVED** [review]
Wonderful, this is all testing well for me now @chad1008!

✅ Works well in a short list view
✅ In long posts and pages the windowing logic appears to be working correctly
✅ In nested lists, the horizontal scrollbar is available
✅ Keyboard behaviour when navigating between tabs and in the list view is working well
✅ Short list view expanding into a long list engages windowing logic correctly
✅ Document outline styling is correct and links work as expected
✅ Works appropriately in the site editor, too

LGTM, thanks again for all the back and forth! ✨

**Comment by chad1008** [comment]
thank you again @andrewserong! This PR was greatly improved by your input. I really appreciate it.

**Comment by andrewserong** [comment]
Any time! Likewise, thanks again for all the iterations here! 🙇
